File: src/main/java/core/language/service/LanguageService.java
package core.language.service;

import core.language.Language;

import java.util.List;

public interface LanguageService {
    List<Language> getLanguages();

    void createLanguage(Language language);

    Language getLanguage(Long id);

    void updateLanguage(Language language);

    void deleteLanguage(Long id);
}

File: src/main/resources/angular/app/language/language.service.js
angular.module('language').service('languageService',
    ['$http', '$q', function ($http) {
        var URL = 'http://localhost:8080/EducationalSystem/api/languages';

        var getAllLanguages = function () {
            return $http.get(URL);
        };

        var getLanguage = function (id) {
            return $http.get(URL + '/' + id);
        };

        var createNewLanguage = function (name) {
            return $http.post(URL, {name: name});
        };

        var deleteLanguage = function (language) {
            return $http.delete(URL + '/' + language.id);
        };

        return {
            getAllLanguages: getAllLanguages,
            getLanguage: getLanguage,
            createNewLanguage: createNewLanguage,
            deleteLanguage: deleteLanguage
        };
    }]);